delivery/src/main/java/com/storyous/delivery/common/DeliveryViewModel.kt

```diff
@@ -34,53 +39,71 @@ class DeliveryViewModel : ViewModel(), CoroutineScope by CoroutineProviderScope(
         const val MESSAGE_OK_DECLINED = 3
     }
 
-    private val selectedOrderIdLive = MutableLiveData<String>(null)
-    private val selectedOrderLive = Transformations.switchMap(selectedOrderIdLive) {
+    private val selectedOrderIdLive = MutableLiveData<String?>(null)
+    var selectedOrderId: String?
+        get() = selectedOrderIdLive.value
+        set(value) {
+            selectedOrderIdLive.value = value
+            Timber.i("Delivery order selected $value")
+        }
+    val selectedOrderLive = Transformations.switchMap(selectedOrderIdLive) {
         it?.let { DeliveryConfiguration.deliveryRepository?.getOrderLive(it) }
             ?: MutableLiveData<DeliveryOrder>(null)
     }
+    val selectedOrder: DeliveryOrder? get() = selectedOrderLive.value
     private val deliveryOrdersLive: LiveData<List<DeliveryOrder>> =
         MediatorLiveData<List<DeliveryOrder>>().apply {
             addSource(DeliveryConfiguration.deliveryRepository!!.deliveryOrdersLive) { orders ->
-                if (getSelectedOrder()?.orderId?.let { orderId -> orders.find { it.orderId == orderId } } == null) {
+                if (selectedOrder?.orderId?.let { orderId -> orders.find { it.orderId == orderId } } == null) {
                     deselectOrder()
                 }
                 value = orders
             }
         }
 
-    val acceptFunction = Transformations.switchMap(selectedOrderLive) {
-        MutableLiveData(false to false).apply {
-            if (it != null) {
-                launch {
-                    value = withContext(provider.IO) {
-                        with(DeliveryConfiguration) {
-                            acceptVisible(it) to acceptEnabled(it)
-                        }
-                    }
+    val acceptFunction = selectedOrderLive.switchMap { order ->
+        liveData {
+            emit(false to false)
+            order?.runCatching {
+                with(DeliveryConfiguration) {
+                    acceptVisible(this@runCatching) to acceptEnabled(this@runCatching)
                 }
+            }?.getOrNull().also {
+                emit(it)
             }
         }
     }
-    val cancelFunction = Transformations.map(selectedOrderLive) {
+    val cancelFunction = selectedOrderLive.map {
         with(it?.state == DeliveryOrder.STATE_NEW) { this to this }
     }
-    val dispatchFunction = Transformations.switchMap(selectedOrderLive) {
-        MutableLiveData(false to false).apply {
-            if (it != null) {
-                launch {
-                    value = withContext(provider.IO) {
-                        DeliveryConfiguration.dispatchVisible(it) to DeliveryConfiguration.dispatchEnabled(
-                            it
-                        )
-                    }
+    val dispatchFunction = selectedOrderLive.switchMap { order ->
```

Comment:
Tady je vlastně třikrát ta samá funkce, ne? Nepůjde ta duplicita nějak odstranit?

Reply:
Taky jsem si to říkal, ale byl jsem línej to sjednotit. Ještě na to kouknu